Components documentation improvements (#2121)
withastro/starlight: docs/src/content/docs/components/link-buttons.mdx

---
title: Link Buttons
description: Learn how to create link buttons in Starlight for visually distinct call-to-action links.
---

import { LinkButton } from '@astrojs/starlight/components';

To display visually distinct call-to-action links, use the `<LinkButton>` component.

import Preview from '~/components/component-preview.astro';

<Preview>

<LinkButton slot="preview" href="/getting-started/">
	Read the docs
</LinkButton>

</Preview>

## Import

```tsx
import { LinkButton } from '@astrojs/starlight/components';
```

## Usage

> Display a link button, useful for directing users to the most relevant or actionable content and often used on landing pages, using the `<LinkButton>` component.

Reviewer: While it means we repeat ourselves slightly between the opening and this section, I think this still flows better so it’s worth doing:
```suggestion
Use the `<LinkButton>` component to display a visually distinct call-to-action link. A link button is useful for directing users to the most relevant or actionable content and is often used on landing pages
```

	</LinkButton>
</Fragment>

</Preview>

### Add icons to cards

Include an icon in a link button using the [`icon`](#icon) attribute set to the name of [one of Starlight’s built-in icons](/reference/icons/#all-icons).

The [`iconPlacement`](#iconplacement) attribute can be used to place the icon before the text by setting it to `start` (defaults to `end`).

<Preview>

```mdx {8-9}
# src/content/docs/example.mdx

import { LinkButton } from '@astrojs/starlight/components';

<LinkButton
	href="https://docs.astro.build"
	variant="secondary"
	icon="external"
	iconPlacement="start"
>
	Related: Astro
</LinkButton>
```

<Fragment slot="markdoc">

```markdoc {6-7}
# src/content/docs/example.mdoc

{% linkbutton
   href="https://docs.astro.build"
	 variant="secondary"
	 icon="external"
	 iconPlacement="start" %}
Related: Astro
{% /linkbutton %}
```

</Fragment>

<LinkButton
	slot="preview"
	href="https://docs.astro.build"
	variant="secondary"
	icon="external"
	iconPlacement="start"
>
	Related: Astro
</LinkButton>

</Preview>

## `<LinkButton>` Props

**Implementation:** [`LinkButton.astro`](https://github.com/withastro/starlight/blob/main/packages/starlight/user-components/LinkButton.astro)

> The `<LinkButton>` component accepts the following props and also any other `<a>` attributes:

Reviewer: I think it would be nice to link here like the `LinkCard` page does:

```suggestion
The `<LinkButton>` component accepts the following props and also any [other `<a>` attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a):
```

### `href`

> **required**

Reviewer: Missing line break here!

```suggestion
**required**<br>
```
